internal-functions

```diff
@@ -75,3 +75,49 @@ fn-acl-check-app() {
     dokku_log_fail "You can only modify ACL using local dokku command on target host"
   fi
 }
+
+fn-acl-check-service() {
+  declare SERVICE_TYPE="$1" SERVICE="$2"
+
+  local SERVICE_PATH="$DOKKU_LIB_ROOT/services/$SERVICE_TYPE/$SERVICE"
+  if ! [[ -d $SERVICE_PATH ]]; then
+    dokku_log_fail "Service $SERVICE of type $SERVICE_TYPE does not exist"
+  fi
+
+  if [[ -n "${NAME:-}" ]]; then
+    dokku_log_fail "You can only modify ACL using local dokku command on target host"
+  fi
+}
+
+check-app-acl() {
+  declare desc="Checks if the current user has an ACL entry for the app"
+  declare APP=$1
+
+  if ! ( verify_app_name "$APP" 2>/dev/null ); then
+    dokku_log_fail "User $SSH_NAME does not have permissions to run $CMD on $APP, or $APP does not exist"
+  fi
+
+  ACL_FILE="$DOKKU_ROOT/$APP/acl/$SSH_NAME"
+  [[ -f "$ACL_FILE" ]] && exit 0
+
+  dokku_log_fail "User $SSH_NAME does not have permissions to run $CMD on $APP, or $APP does not exist"
+}
+
+check-service-acl() {
+  declare desc="Checks if the current user has an ACL entry for the service"
+  declare CMD=$1
+  declare SERVICE=$2
```

Comment:
Declarations of parameters are normally done on a single line like so:

```
declare CMD="$1" SERVICE="$2"
```

They are also quoted to guard against expansion.